lib/core/public/run.js

```diff
@@ -112,28 +112,32 @@ axe.run = function (context, options, callback) {
 		});
 	}
 
-	axe._runRules(context, options, function (rawResults) {
+	axe._runRules(context, options, function (rawResults, cleanup) {
 		let respond = function (results) {
-			try {
+			// Cleanup BEFORE executing callback / promise to prevent interfering
+			// with a potential future axe.run() call.
+			cleanup();
+			if (callback !== noop) {
 				callback(null, results);
-			} catch(e) {
-				axe.log(e);
+			} else {
+				resolve(results);
 			}
-			resolve(results);
 		};
 		if (options.performanceTimer) {
 			axe.utils.performanceTimer.end();
 		}
 
-		try {
+		let results;
+		// try {
```

Comment:
Why are you commenting this out? Maybe we should remove it if it's not used/needed.

Reply:
My bad, committed something I shouldn't have :/